### ISSUE TITLE
JavaScript Code Injection

### ISSUE THREAD

**Issue by biggeeves** [vanderbilt/redcap-external-modules#58]
Using the a field type "text" in the config.json file allows entry of javascript, which gets rendered upon saving.  

Reproduce:
Add `<script>alert('hi')</script>` as the setting for a text field, press save, and an a javascript alert pops up

It is possible that a malicious user could inject code which other users might execute without knowing it. 

Solution:
Sanitize all inputs.

Precedence: It is already done for text fields in instruments.  It should be done for all text fields in External Module Settings. 

Possible Exception: Create a field type "script" which is available to super users only.

**Comment by 123andy** [comment]
It is a good point, but there needs to be a way to permit javascript from EM's too.  So, perhaps two different field types makes the most sense...
-A

**Comment by mmcev106** [comment]
I think I may need more information to help with this.  I tried inserting a javascript alert as described on a few of our modules, but it was not executed.  I could see this happening if a module was inserting a setting value into the DOM, but I don't know of a module off the top of my head that does that in order to think through the use case.  Any information related to goals(s) for inserting javascript in setting values might be helpful as well.  I don't believe I've encountered a need to do that before.
